src/ast.h

```diff
@@ -0,0 +1,299 @@
+/*
+ * (ast.h | 8 Mar 19 | Amr Anwar)
+ * 
+ * The abstract syntax trees structs
+ * 
+*/
+
+#ifndef ast_h
+#define ast_h
+
+#include <stdint.h>
+
+#include "list.h"
+#include "lexer.h"
+
+typedef struct expr expr;
+typedef struct pattern pattern;
+
+typedef struct {
+    list *stmts;
+} piece;
+
+typedef enum {
+    let_stmt_type,
+    fn_stmt_type,
+    ret_stmt_type,
+    expr_stmt_type,
+    fixed_stmt_type
+} stmt_type;
+
+typedef enum {
+    BREAK,
+    CONTINUE,
+} fixed_stmt;
+
+typedef struct {
+    char* ident;
+    list* patts;
+} param_list;
+
+typedef struct {
+    char *n_ident;
+    list *param_list;
```

Comment:
this is error.
`param_list` is a type, you defined by `typedef`.
remove the param_list struct.